src/pkg/exporter/cache.go

```diff
@@ -0,0 +1,88 @@
+package exporter
+
+import (
+	"sync"
+	"time"
+)
+
+var c *cache
+
+type cachedValue struct {
+	Value      interface{}
+	Expiration int64
+}
+
+type cache struct {
+	CacheTime int64
+	store     map[string]cachedValue
+	*sync.RWMutex
+}
+
+// CacheGet get a value from cache
+func CacheGet(key string) (value interface{}, ok bool) {
```

Comment:
This is not a blocker but ideally you should limit the accepted keys, otherwise the cache can grow endlessly by current design.